src/pages/videoCategories/videoCategories.js

```diff
@@ -1,39 +1,105 @@
 import React, { useEffect, useState, useMemo } from "react";
 import { useParams } from "react-router-dom";
+import { Box, Button, Modal, Typography } from "src/components";
+import { LinkButton } from "src/components/Button";
+import userService from "src/services/userService";
 import videoService from "src/services/videoService";
 import Video from "../components/video";
 import { Wrapper } from "./videoCategories.styled";
 
 export default function VideoCategories() {
   const { categoryName } = useParams();
   const [videos, setVideos] = useState([]);
+  const [playlists, setPlaylists] = useState([]);
+  const [showPlaylist, setShowPlaylist] = useState(false);
+  const [video, setVideo] = useState(null);
 
   useEffect(() => {
     getAllVideos();
+    getPlaylists();
   }, []);
 
   const categoryVideos = useMemo(() => {
     return videos.filter((video) => video.categoryName === categoryName);
   }, [videos, categoryName]);
 
+  const addToPlaylist = (video) => {
+    setVideo(video);
+    setShowPlaylist(true);
+  };
+
+  const addVideoToList = ({ _id }) => {
+    userService
+      .addVideoToPlaylist({ _id, video })
+      .then(() => {
+        setShowPlaylist(false);
+      })
+      .finally(() => {
+        setShowPlaylist(false);
+      });
+  };
```

Comment:
good that you have used finally here but should add catch block as well to handle errors.